src/ioctl.rs

```diff
@@ -154,17 +198,104 @@ type IoctlRequest = c_int;
 #[cfg(not(target_env = "musl"))]
 type IoctlRequest = c_ulong;
 
-/// Run an ioctl with no arguments.
+/// Run an [`ioctl`](http://man7.org/linux/man-pages/man2/ioctl.2.html)
+/// with no arguments.
+///
+/// # Arguments
+///
+/// * `fd`: an open file descriptor corresponding to the device on which
+/// to call the ioctl.
+/// * `req`: a device-dependent request code.
+///
+/// # Safety
+///
+/// The caller should ensure to pass a valid file descriptor and have the
+/// return value checked.
+///
+/// # Examples
+///
+/// ```
+/// # extern crate libc;
+/// #[macro_use] extern crate vmm_sys_util;
+/// #
+/// # use libc::{open, O_CLOEXEC, O_RDWR};
+/// # use std::fs::File;
+/// # use std::os::raw::{c_char, c_uint};
+/// # use std::os::unix::io::FromRawFd;
+/// use vmm_sys_util::ioctl::ioctl;
+///
+/// let open_flags = O_RDWR | O_CLOEXEC;
+/// let kvm_fd = unsafe { open("/dev/kvm\0".as_ptr() as *const c_char, open_flags) };
+/// const KVMIO: c_uint = 0xAE;
```

Comment:
nit: move the `const` declarations separately from the fd/ioctl logic, to neatly delimit the functional blocks 
```rust
#[macro_use] extern crate vmm_sys_util;
use vmm_sys_util::ioctl::ioctl;

const KVMIO: c_uint = 0xAE;
const KVM_API_VERSION: u32 = 12;
ioctl_io_nr!(KVM_GET_API_VERSION, KVMIO, 0x00);

let open_flags = O_RDWR | O_CLOEXEC;
let kvm_fd = unsafe { open("/dev/kvm\0".as_ptr() as *const c_char, open_flags) };

let ret = unsafe { ioctl(&File::from_raw_fd(kvm_fd), KVM_GET_API_VERSION()) };

assert_eq!(ret as u32, KVM_API_VERSION);
```

This applies to all the ioctl examples, I won't post individual nit comments with code suggestions.

Reply:
Comments are addressed, thanks!